toolsrc/src/vcpkg/commands.list.cpp

```diff
@@ -4,12 +4,51 @@
 #include <vcpkg/commands.h>
 #include <vcpkg/help.h>
 #include <vcpkg/vcpkglib.h>
+#include <vcpkg/base/json.h>
 
 namespace vcpkg::Commands::List
 {
     static constexpr StringLiteral OPTION_FULLDESC =
         "--x-full-desc"; // TODO: This should find a better home, eventually
 
+    static constexpr StringLiteral OPTION_JSON = "--json";
+
+    static void do_print_json(std::vector<const vcpkg::StatusParagraph*> installed_packages)
+    {
+        Json::Object obj;
+
+        for (const StatusParagraph* status_paragraph : installed_packages)
+        {
+            auto current_spec = status_paragraph->package.spec;
+            if (!obj.contains(current_spec.to_string()))
```

Comment:
```suggestion
            if (auto library_obj = obj.get(current_spec))
```

then swap the logic.